// mern.common/dist/functions/response.send.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponseMessage = exports.HTTPStatusCodes = void 0;
exports.SendResponse = SendResponse;
var HTTPStatusCodes;
(function (HTTPStatusCodes) {
    HTTPStatusCodes[HTTPStatusCodes["OK"] = 200] = "OK";
    HTTPStatusCodes[HTTPStatusCodes["CREATED"] = 201] = "CREATED";
    HTTPStatusCodes[HTTPStatusCodes["VERIFIED"] = 202] = "VERIFIED";
    HTTPStatusCodes[HTTPStatusCodes["BAD_REQUEST"] = 400] = "BAD_REQUEST";
    HTTPStatusCodes[HTTPStatusCodes["UNAUTHORIZED"] = 401] = "UNAUTHORIZED";
    HTTPStatusCodes[HTTPStatusCodes["FORBIDDEN"] = 403] = "FORBIDDEN";
    HTTPStatusCodes[HTTPStatusCodes["NOT_FOUND"] = 404] = "NOT_FOUND";
    HTTPStatusCodes[HTTPStatusCodes["CONFLICT"] = 409] = "CONFLICT";
    HTTPStatusCodes[HTTPStatusCodes["INTERNAL_SERVER_ERROR"] = 500] = "INTERNAL_SERVER_ERROR";
    HTTPStatusCodes[HTTPStatusCodes["BAD_GATEWAY"] = 502] = "BAD_GATEWAY";
})(HTTPStatusCodes || (exports.HTTPStatusCodes = HTTPStatusCodes = {}));
var ResponseMessage;
(function (ResponseMessage) {
    ResponseMessage["SUCCESS"] = "Operation successful";
    ResponseMessage["CREATED"] = "Resource successfully created";
    ResponseMessage["VERIFIED"] = "Resource verified";
    ResponseMessage["BAD_REQUEST"] = "Bad request";
    ResponseMessage["UNAUTHORIZED"] = "Unauthorized access";
    ResponseMessage["FORBIDDEN"] = "Access forbidden";
    ResponseMessage["NOT_FOUND"] = "Resource not found";
    ResponseMessage["INTERNAL_SERVER_ERROR"] = "Internal server error";
    ResponseMessage["BAD_GATEWAY"] = "Bad gateway";
})(ResponseMessage || (exports.ResponseMessage = ResponseMessage = {}));
function SendResponse(res, status, message, data) {
    try {
        return res.status(status)
            .json(Object.assign({ message }, (data !== undefined && data !== null && data)));
    }
    catch (err) {
        throw new Error(err);
    }
}
;
